### ISSUE TITLE
[studio] Clustering - Error while updating published repo for site  

### ISSUE THREAD

**Issue by alhambrav** [craftercms/craftercms#2913]
## Describe the bug
There's sometimes an error while updating published repo for site when setting up a second node or creating another site from one of the nodes

## To Reproduce
1. Setup a Studio for clustering on node a and create a site `sitea`
2. Setup a second studio for clustering on node b and watch the logs on node a for errors like below
```
[ERROR] 2019-03-06T09:57:59,458 [Studio_Clustering_Thread_Pool5] [cluster.StudioNodeSyncPublishedTask] | Error while updating published repo for site sitea
```
3. Create a site `siteb` on node b and just watch the logs
4. You can also setup another node, node c, then create a site `sitec` and watch the logs for node a and node b, which might display something like below:
```
[ERROR] 2019-03-05T14:02:29,306 [Studio_Clustering_Thread_Pool7] [cluster.StudioNodeSyncPublishedTask] | Error while updating published repo for site sitec
[ERROR] 2019-03-05T14:03:28,931 [Studio_Clustering_Thread_Pool5] [cluster.StudioNodeSyncPublishedTask] | Error while updating published repo for site sitec
[ERROR] 2019-03-05T14:04:29,185 [Studio_Clustering_Thread_Pool10] [cluster.StudioNodeSyncPublishedTask] | Error while updating published repo for site sitec
[ERROR] 2019-03-05T14:05:28,962 [Studio_Clustering_Thread_Pool10] [cluster.StudioNodeSyncPublishedTask] | Error while updating published repo for site sitec
[ERROR] 2019-03-05T14:06:29,081 [Studio_Clustering_Thread_Pool9] [cluster.StudioNodeSyncPublishedTask] | Error while updating published repo for site sitec
[ERROR] 2019-03-05T14:07:28,991 [Studio_Clustering_Thread_Pool6] [cluster.StudioNodeSyncPublishedTask] | Error while updating published repo for site sitec
[ERROR] 2019-03-05T14:08:29,044 [Studio_Clustering_Thread_Pool5] [cluster.StudioNodeSyncPublishedTask] | Error while updating published repo for site sitec
```

## Expected behavior
There should be no errors when a new site is created in another node

## Logs
Node A: 192.168.1.3 (First node up, site created sitea)
Node B: 192.168.1.18 (Second node up, site created siteb)
Node C: 192.168.1.15 (Third node up, site created sitec)

Node A logs after creating "sitec" on node C: https://gist.github.com/alhambrav/64503714875382b7cc58c9d5fbc2e91f
Node B logs after creating "sitec" on node C:
https://gist.github.com/alhambrav/4fea5122e5933affb6ec74a1983b6f7e
Node C logs after creating "sitec" on node C:
https://gist.github.com/alhambrav/b26240614d8e486c4ffa93b05a59ba0a

Notice the `[ERROR] 2019-03-05T14:08:29,044 [Studio_Clustering_Thread_Pool5] [cluster.StudioNodeSyncPublishedTask] | Error while updating published repo for site sitec
`
### Version
Studio Version Number: 3.1.0-SNAPSHOT-fe961b
Build Number: fe961b70b2a237c90e3d5ab7367cca72d5949d4c
Build Date/Time: 03-06-2019 09:44:20 -0500

### OS
OS X, Ubuntu

### Browser
Chrome browser

## Additional context
I spoke with @dejan-brkic and he says there might be some race conditions or something that occurs when we see the error while updating published repo

**Comment by alhambrav** [comment]
Milestone and priority changed based on meeting with @dejan-brkic and @sumerjabri .  Please update the log message to be a little bit more detailed.

**Comment by dejan-brkic** [comment]
Can not reproduce 

```
Studio Version Number: 3.1.1-SNAPSHOT-45408d

Build Number: 45408d52ece705a6a2d24f4aeb9fbaee56195e56
```

**Comment by alhambrav** [comment]
I see the error message `Error while updating published repo for site` when starting up another node.  Here a re the steps:
1. Setup Studio for clustering
2. Startup Studio, this will be node a, then create a site, `sitea`
3. Setup another Studio for clustering
4. Startup Studio, this will be node b.  Note the logs in node a.
I see the following message in node a when I started up the second node:
```
[ERROR] 2019-05-29T13:44:16,554 [Studio_Clustering_Thread_Pool9] [cluster.StudioNodeSyncPublishedTask] | Error while updating published repo for site sitea
```
5. Setup yet another Studio for clustering
6. Start Studio, this will be node c. Notice the logs on node a
I see the following message in node a when I started up a third node: 
```
[ERROR] 2019-05-29T14:46:29,095 [Studio_Clustering_Thread_Pool3] [cluster.StudioNodeSyncPublishedTask] | Error while updating published repo for site sitea
[ERROR] 2019-05-29T14:46:29,095 [Studio_Clustering_Thread_Pool6] [cluster.StudioNodeSyncPublishedTask] | Error while updating published repo for site siteb
```

Is this the expected behavior whenever a new node comes along @dejan-brkic ?

**Studio Version**
Studio Version Number: 3.1.1-SNAPSHOT-f64209

Build Number: f642095a15b002e33f08f30816c048e923aaa333
Build Date/Time: 05-29-2019 10:16:24 -0400
